scripts/cli/handler/parse.py

```diff
@@ -0,0 +1,377 @@
+import datetime
+import glob
+import os
+import re
+import shutil
+import subprocess
+import sys
+import time
+import signal
+
+from os.path import join
+
+from util.configfilehandler import ConfigFileHandler
+from util import util
+
+def __current_time():
+  '''This function returns the current date and time as string.'''
+  return datetime.datetime.fromtimestamp(
+    time.time()).strftime('%Y-%m-%d %H:%M:%S')
+
+def __handle_parser_crash(projectdir):
+  #--- Get line containing crash message in the log file ---#
+
+  with open(join(projectdir, 'parselog'), 'r') as f:
+    for line in f:
+      if 'CODECOMPASS CRASH' in line:
+        crash_message = line
+
+  #--- Check crashed build action ---#
+
+  if not crash_message or 'UNKNOWN' in crash_message:
+    sys.stderr.write('Crash caused by an unknown action! Aborting!\n')
+    sys.exit(1)
+
+  try:
+    m = re.match(r'CODECOMPASS CRASH ON ACTION (\d*)', crash_message)
+    build_action = int(m.group(1))
+  except:
+    sys.stderr.write('Crash caused by an unknown action (malformed crash\
+      message)! Aborting!\n')
+    sys.exit(1)
+
+  print 'Crash caused by action ' + str(build_action)
+
+  #--- Manage core files ---#
+
+  core_count = len(filter(lambda x: 'crash_' in x, os.listdir(projectdir)))
+
+  coreFilename = next(iter(glob.glob(join(projectdir, 'core*'))), None)
+  dummyFilename = next(iter(glob.glob(join(projectdir, 'dummy.core*'))), None)
+
+  if dummyFilename:
+    if not coreFilename:
+      coreFilename = dummyFilename
+    else:
+      os.remove(dummyFilename)
+
+  if coreFilename:
+    print 'A core dump was found as ', coreFilename
+
+    if core_count > 5 and not coreFilename.startswith('dummy.core'):
+      print 'Too many saved core dumps. I just delete this one...'
+      os.remove(coreFilename)
+    else:
+      os.rename(coreFilename, join(projectdir, 'crash_' + str(build_action)))
```

Comment:
Dummy-core was a special core file created in the Earhart which enabled us to properly see a